infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/SessionManager.java

```diff
@@ -581,5 +626,29 @@ public boolean isConnectionReceiver() {
     public void notifyLeadershipChange() {
         incomingMsgHandler.leadershipChanged();
     }
+
+    public void sendOutSessionToSinkSide(LogReplication.ReplicationSubscriber subscriber) {
+        Set<LogReplicationSession> sessionForSinkSide = this.createOutgoingSessionsBySubscriber(subscriber);
+        for (LogReplicationSession session : sessionForSinkSide) {
+            LogReplication.LogReplicationSinkSessionInitializationMsg msg = LogReplication.LogReplicationSinkSessionInitializationMsg.newBuilder().setSession(session).build();
+            CorfuMessage.RequestPayloadMsg payload = CorfuMessage.RequestPayloadMsg.newBuilder().setLrSinkSessionInitialization(msg).build();
+
+            updateRouterWithNewSessions();
+            createSourceFSMs();
+            logNewlyAddedSessionInfo();
+            if (topology.getRemoteClusterEndpoints().containsKey(session.getSourceClusterId())) {
+                // initiate connection to remote Sink cluster
+                router.connect(topology.getAllClustersInTopology().get(session.getSinkClusterId()), session);
+            } else {
+                // initiate connection to remote Source cluster
+                router.connect(topology.getAllClustersInTopology().get(session.getSourceClusterId()), session);
+            }
+            try {
```

Comment:
this block should be in `IRetry` block.
This is a good example of it: https://github.com/CorfuDB/CorfuDB/blob/master/infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java#L433

It would look something like:

IRetry.build(ExponentialBackoffRetry.class, () -> {
                try {
                    lCompletableFuture<LogReplication.LogReplicationMetadataResponseMsg> cf = router.sendRequestAndGetCompletable(session, payload, router.getSessionToRemoteClusterDescriptor().get(session).getClusterId());
                    cf.get(CorfuLogReplicationRuntime.DEFAULT_TIMEOUT, TimeUnit.MILLISECONDS);
                } catch (TimeoutException e) {
                    log.error("Caught exception while waiting for ACK for <rpc>. Retry.", e);
                    throw new RetryNeededException();
                }
                return null;
            }